src/main/webapp/js/reducers/entity-reducer.js

```diff
@@ -107,11 +112,21 @@ export default (state = initialState, action) => {
       });
     case UPDATE_ENTITY:
       return update(state, {
-        [entityType]: { $set: { [payload.data.id]: payload.data } },
+        [entityType]: {
+          $set: _.map(payload.data, elem => ({
```

Comment:
it should be:
`{ $set: { [payload.data.id]: _.map(payload.data, elem => ({
            ...elem,
            extraFields: _.keyBy(_.get(elem, 'extraFields', {}), 'name'),
          }))
 } },`